components/loading/LoadingBar.vue

```diff
@@ -0,0 +1,59 @@
+<template>
+  <v-app ref="loaderContainer" class="v-app">
+    <v-progress-linear
+      :active="loading"
+      :indeterminate="indeterminate"
+      :height="4"
+      :bottom="true"
+      :rounded="true"
+      :color="barColor"
+      :value="barValue"
+    ></v-progress-linear>
+  </v-app>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+
+let timeout: number;
+export default Vue.extend({
+  data() {
+    return {
+      loading: false,
+      barColor: 'primary',
+      indeterminate: true,
+      barValue: 0
+    };
+  },
+  methods: {
+    start() {
+      window.clearTimeout(timeout);
+      this.loading = true;
+      this.indeterminate = true;
+      this.barColor = 'primary';
+      this.barValue = 0;
+    },
+    finish() {
+      this.indeterminate = true;
+      this.barColor = 'primary';
+      this.barValue = 0;
+      // Ensure that the 'slide up' animation appears after loading finishes. Without this
+      // the animation could be skipped due to DOM blocking.
+      timeout = window.setTimeout(() => (this.loading = false), 50);
+    },
+    fail() {
+      this.barColor = 'red';
+      this.indeterminate = false;
+      this.barValue = 30;
```

Comment:
Why 30?

Seems like it'd be a perfect use of "indeterminate", or even put it at 100 so it covers the entire width?.

Reply:
@mrtimscampi Indeterminate doesn't make sense imo, as the progressbar will keep "flowing" in the event of an error, which doesn't make sense imo, as when an error loading happens it usually means that the loading process was halted, thus, no progress is happening. 

30 fills a quarter of the bar while being still visible with the sidebar open.

Reply:
After testing it, I think the sidebar shouldn't be hiding the loading bar (in desktop mode, at least).

The sidebar isn't closable on desktop (it was before, but it's not intended to be in the current design), so it looks a bit weird there. And on mobile, you'll never be able to navigate while the navigation bar is open (It should close when you click on a link).

As for the error bar, I think filling it completely would be better.  
I expect that version to barely be shown, since we should handle errors in a more verbose way. We have an error page callable from everywhere, which should be used when API calls fail. For server disconnections, I'm planning to have an overlay open with a re-connection attempt tracker when the websocket goes down. So it'll only be shown in some very rare cases where we don't handle things properly, but then 30% of the bar filled doesn't make sense. It should be as visible as possible, hence the 100% value. At least imo :smiley: